sycl/include/sycl/ext/oneapi/experimental/annotated_arg/annotated_arg.hpp

```diff
@@ -383,6 +362,77 @@ __SYCL_TYPE(annotated_arg) annotated_arg<T, detail::properties_t<Props...>> {
   R operator<<(const annotated_arg<T2, PropertyList2> &other) const {
     return obj << other.obj;
   }
+
```

Comment:
```suggestion
  // *************************************************************************
  // All static error checking is added here instead of placing inside neat
  // functions to minimize the number lines printed out when an assert
  // is triggered
  // *************************************************************************
```